### PR TITLE
[docs] getting-started-guide

### LINKED ISSUE #15: docs: developers getting-started guide
**Python Version** (output of `python --version`): 2.7.*

**Development or Deployment Environment?**: Any

**Release Tag or Master**: master

**Expected Behavior**: 
We need to generate docs on how to set up the an armada environment. 
- [x] create docs dir
- [x] create development dir with a file `getting-started.rst`
- [x] add a section for development with virtualenv

### PR DESCRIPTION
**Pull request by gardlt**
* updated README.md
* added getting started guide

**What is the purpose of this pull request?**:

Instructions to set up an development environment. 

**What issue does this pull request address?**: Fixes #15

**Notes for reviewers to consider**:

**Specific reviewers for pull request**:
@alanmeadows @v1k0d3n

### CODE REVIEW

File: docs/development/getting-started.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+   pip install pygit2==0.25.0
+   pip install -r requirements.txt
+   pip install -r test-requirements.txt
+
+
+
+.. note:: this will install latest libgit2 library so you have to make sure you install the same version library with pip ( current version: 0.25.0 )
+
+
+Virtualenv
```
Comment (v1k0d3n) [inline docs/development/getting-started.rst]: I would strongly suggest _just_ to use virtualenv, and install the others directly only if really needed. Secondly, I would suggest that the main use to be from the Docker container. If lots of options are provided, then lots of options will need to be supported. Limiting the scope early would be my recommendation.

File: docs/development/getting-started.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+.. note:: When using Halcyon it will not generate a config file. Run the following commands to create one:
+
+.. code-block:: bash
+
+   kubectl config set-credentials myself --username=admin --password=secret
+   kubectl config set-cluster local-server --server=http://localhost:8080
+   kubectl config set-context default-context --cluster=local-server --user=myself
+   kubectl config use-context default-context
+   kubectl config set contexts.default-context.namespace the-right-prefix
+   kubectl config view
```
Comment (v1k0d3n) [inline docs/development/getting-started.rst]: [There is a mechanism for Halcyon to provide credentials](https://github.com/att-comdev/halcyon-vagrant-kubernetes/blob/master/get-k8s-creds.sh).

Comment (gardlt) [inline docs/development/getting-started.rst]: awesome. I will make a reference to this.

File: docs/development/getting-started.rst (hunk 1 above)
Comment (gardlt) [inline docs/development/getting-started.rst]: I see, i will be sure to clean and limit the scope. I can add docker section to this file with some general info.

**Comment by v1k0d3n** [comment]
better. does the docker container work as advertised now?

**Comment by gardlt** [comment]
not currently, the docker container will need a little bit more work this is a current placeholder for when the container work is done ore needs to be updated.

File: docs/development/getting-started.rst [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    LIBGIT_VERSION='0.25.0'
+
+    wget https://github.com/libgit2/libgit2/archive/v${LIBGIT_VERSION}.tar.gz
+    tar xzf v${LIBGIT_VERSION}.tar.gz
+    cd libgit2-${LIBGIT_VERSION}/
+    cmake . -DCMAKE_INSTALL_PREFIX=$LIBGIT2
+    make
+    sudo make install
+    export LDFLAGS="-Wl,-rpath='$LIBGIT2/lib',--enable-new-dtags $LDFLAGS"
+    sudo pip install pygit2==0.25.0
```
Comment (alanmeadows) [inline docs/development/getting-started.rst]: Shouldn't this be `pip install pygit2==$LIBGIT_VERSION`

Comment (gardlt) [inline docs/development/getting-started.rst]: oh yeah, good catch.

**Comment by v1k0d3n** [comment]
LGTM now @gardlt

**Comment by gardlt** [comment]
@alanmeadows @v1k0d3n can we merge this?

**Comment by v1k0d3n** [comment]
Looks like you have taken care of @alanmeadows ask @gardlt. I can merge this in for you.

**Pull request merged**